### PR TITLE
Fix new search showing old results

### PR DESCRIPTION
**Pull request by mjaseem**
On search results view, if you start a new search by entering a new
query, page will reload and but will show the old results.
Fixing this by forcing refresh and clearing old results.

### CODE REVIEW

**Comment by mjaseem** [comment]
This fixes  #1218, but lead to a new issue : on the second search, results page gets scrolled to the bottom and more results keep loading continuously. Any idea why this could happen?

**Comment by mjaseem** [comment]
The issue mentioned above has been fixed and I have amended the commit. 
@veyndan Please review.

**Comment by mjaseem** [comment]
@Meisolsson Please review

File: app/src/main/java/com/github/pockethub/android/ui/search/SearchRepositoryListFragment.java [hunk 1]
```diff
@@ -89,10 +90,16 @@ public void onAttach(Context context) {
...
+    public void forceRefresh(){
+        start();
+        super.forceRefresh();
+    }
+
     @Override
     public void refresh() {
         start();
-        super.refresh();
+        super.refresh(true);
```
Comment (Meisolsson) [inline app/src/main/java/com/github/pockethub/android/ui/search/SearchRepositoryListFragment.java]: We should not force all of the refreshes

File: app/src/main/java/com/github/pockethub/android/ui/search/SearchUserListFragment.java [hunk 2]
```diff
@@ -90,11 +92,22 @@ public void onAttach(Context context) {
...
+    public void forceRefresh(){
+        query = getStringExtra(QUERY);
+        super.forceRefresh();
+    }
     @Override
     public void refresh() {
         query = getStringExtra(QUERY);
 
-        super.refresh();
+        super.refresh(true);
```
Comment (Meisolsson) [inline app/src/main/java/com/github/pockethub/android/ui/search/SearchUserListFragment.java]: We should not force all of the refreshes

File: app/src/main/java/com/github/pockethub/android/ui/search/SearchUserListFragment.java [hunk 3]
```diff
@@ -90,11 +92,22 @@ public void onAttach(Context context) {
...
+    }
+
+    @Override
+    protected void onDataLoaded(List<Item> newItems) {
+        items.clear();
```
Comment (Meisolsson) [inline app/src/main/java/com/github/pockethub/android/ui/search/SearchUserListFragment.java]: This should not be done, it doesn't allow the list to load more items.

File: app/src/main/java/com/github/pockethub/android/ui/search/SearchRepositoryListFragment.java [hunk 4]
```diff
@@ -149,6 +156,12 @@ private boolean openRepositoryMatch(final String query) {
         return true;
     }
 
+    @Override
+    protected void onDataLoaded(List<Item> newItems) {
+        items.clear();
```
Comment (Meisolsson) [inline app/src/main/java/com/github/pockethub/android/ui/search/SearchRepositoryListFragment.java]: This should not be done, it doesn't allow the list to load more items.

**Comment by mjaseem** [comment]
@Meisolsson I have made the requested changes

**Pull request merged**